### PR TITLE
test: add e2e test for usdt special case

### PR DESCRIPTION
**Pull request by just-toby**
## Description

adds an automated e2e test for the special case of USDT mainnet requiring allowance revocation before increasing the allowance to the required minimum

### Automated testing

- [x] Integration/E2E test

### CODE REVIEW

**Comment by just-toby** [comment]
Current dependencies on/for this PR:
* main
  * **PR #6726** <a href="https://app.graphite.dev/github/pr/Uniswap/interface/6726" target="_blank">Graphite</a> 
    * **PR #6727** <a href="https://app.graphite.dev/github/pr/Uniswap/interface/6727" target="_blank">Graphite</a> 
      * **PR #6728** <a href="https://app.graphite.dev/github/pr/Uniswap/interface/6728" target="_blank">Graphite</a> 
        * **PR #6729** <a href="https://app.graphite.dev/github/pr/Uniswap/interface/6729" target="_blank">Graphite</a> 
          * **PR #6730** <a href="https://app.graphite.dev/github/pr/Uniswap/interface/6730" target="_blank">Graphite</a> 
            * **PR #6784** <a href="https://app.graphite.dev/github/pr/Uniswap/interface/6784" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/Uniswap/interface/6784?utm_source=stack-comment).

File: src/hooks/useTokenAllowance.ts [hunk 1]
```diff
@@ -48,7 +48,8 @@ export function useUpdateTokenAllowance(
       if (!contract) throw new Error('missing contract')
       if (!spender) throw new Error('missing spender')
 
-      const allowance = MaxUint256.toString()
+      const maxAllowance = MaxUint256.toString()
```
Comment (zzmp) [inline src/hooks/useTokenAllowance.ts:51]: This is static. Could you bring it to the top of the file as `const MAX_ALLOWANCE = MaxUint256.toString()`?

File: cypress/e2e/permit2.test.ts [hunk 2]
```diff
@@ -85,7 +85,45 @@ describe('Permit2', () => {
...
+      expectTokenAllowanceForPermit2ToBeMax(DAI)
+
+      // Verify transaction
+      cy.wait('@eth_sendRawTransaction')
+      cy.hardhat().then((hardhat) => hardhat.mine())
+      cy.contains('Success')
+      cy.get(getTestSelector('popups')).contains('Swapped')
+    })
+
+    it('swaps USDT with existing permit, and existing but insufficient token approval', () => {
```
Comment (zzmp) [inline cypress/e2e/permit2.test.ts:97]: Add a comment stating why USDT needs a special test case.

File: cypress/e2e/permit2.test.ts [hunk 3]
```diff
@@ -85,7 +85,45 @@ describe('Permit2', () => {
...
+      cy.hardhat().then(async (hardhat) => {
+        // fund with 2 USDT
+        await hardhat.fund(hardhat.wallet, CurrencyAmount.fromRawAmount(USDT, 2e6))
+        await hardhat.mine()
+        // set token allowance to 1 USDT
+        await hardhat.approval.setTokenAllowanceForPermit2({ owner: hardhat.wallet, token: USDT }, 1e6)
+        await hardhat.mine()
+        // set permit2 allowance to max
+        await hardhat.approval.setPermit2Allowance({ owner: hardhat.wallet, token: USDT })
+        await hardhat.mine()
```
Comment (zzmp) [inline cypress/e2e/permit2.test.ts:107]: You only need to mine once.

nit: Comments are just re-iterating the test case, so I don't think they are necessary.

```suggestion
        await hardhat.fund(hardhat.wallet, CurrencyAmount.fromRawAmount(USDT, 2e6))
        await hardhat.approval.setTokenAllowanceForPermit2({ owner: hardhat.wallet, token: USDT }, 1e6)
        await hardhat.approval.setPermit2Allowance({ owner: hardhat.wallet, token: USDT })
        await hardhat.mine()
```

**Review by zzmp — CHANGES_REQUESTED** [review]
lgtm, minor change requested

File: cypress/e2e/permit2.test.ts (hunk 3 above)
Comment (just-toby) [reply cypress/e2e/permit2.test.ts:107]: this is what i thought too, but the test fails when i use only one `hardhat.mine()`

**Review by JackShort — APPROVED** [review]

**Review by zzmp — APPROVED** [review]

**Pull request merged**